Cargo.toml

```diff
@@ -41,6 +41,7 @@ derivative = { version = "2.2.0" }
 directories = { version = "5.0.1" }
 dunce = { version = "1.0.4" }
 either = { version = "1.9.0" }
+encoding_rs_io = { version = "0.1.7" }
```

Comment:
Could you expand your reasoning for using `encoding_rs` (shamelessly adding his own crates ;) ) instead of implementing a "simple" BOM check to determine UTF-16 and using `String::from_utf16`?

Reply:
`encoding_rs` isn't mine, but `encoding_rs_io` is. The latter puts an automatic BOM sniffing streaming reader on top of the former.

I suppose since we don't actually need streaming here, so this could be done more simply. Note though that `encoding_rs_io` will also seamlessly handle the UTF-8 BOM, which is not uncommon on Windows AFAIK. (Handling the UTF-8 BOM is basically just a matter of detecting it and then dropping it.)

I'll consider just using `std` for this. Honestly we have so many dependencies (including a dependency on `encoding_rs`) that I don't feel too bad about adding another...

Reply:
Yeah it's really about supporting streaming. That's the main win of `encoding_rs_io`.

Reply:
I looked into this, and it turns out the [UTF-16LE and UTF-16BE APIs](https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf16be) are actually unstable. Do'h. That means doing this correctly is a fair bit more work than just checking the BOM and calling the right routine. We'd have to flip the byte order for an endian mismatch.

`encoding_rs_io` is a small battle tested dependency and it does the right thing for us automatically here, so I'm going to keep it.